src/include/miopen/timer.hpp

```diff
@@ -73,6 +73,29 @@ class CompileTimer
     }
 };
 
+class FunctionTimer{
+ public:
+  FunctionTimer(std::string name)
+      : m_name(std::move(name)),
+        m_beg(std::chrono::high_resolution_clock::now()) { }
+  ~FunctionTimer() {
+    auto end = std::chrono::high_resolution_clock::now();
+    auto dur = std::chrono::duration_cast<std::chrono::microseconds>(end - m_beg);
+    std::cout << m_name << " : " << dur.count() << " musec\n";
```

Comment:
[Question] This seems unusable in MT/MP apps (as the reader needs some means to distinguish one thread/process from another thread/process; see `MIOPEN_ENABLE_LOGGING_MPMT`). Is it Ok for you?